library/core/src/slice/mod.rs

```diff
@@ -662,7 +662,7 @@ impl<T> [T] {
         // SAFETY: the caller must uphold most of the safety requirements for `get_unchecked`;
         // the slice is dereferenceable because `self` is a safe reference.
         // The returned pointer is safe because impls of `SliceIndex` have to guarantee that it is.
-        unsafe { &*index.get_unchecked(self) }
+        unsafe { &*index.get(self).unwrap_unchecked() }
```

Comment:
Would we get the same result by changing SliceIndex::get_unchecked instead of just the slice method?

Reply:
Probably? Will check.

Reply:
We can change that, so that `get` does the full logic and `get_unchecked` does the change from this PR